icechunk/src/storage/s3.rs

```diff
@@ -308,6 +308,135 @@ impl S3Storage {
         Ok(())
     }
 
+    async fn put_object_multipart<
+        I: IntoIterator<Item = (impl Into<String>, impl Into<String>)>,
+    >(
+        &self,
+        settings: &Settings,
+        key: &str,
+        content_type: Option<impl Into<String>>,
+        metadata: I,
+        storage_class: Option<&String>,
+        bytes: &Bytes,
+    ) -> StorageResult<()> {
+        let mut multi = self
+            .get_client()
+            .await
+            .create_multipart_upload()
+            // We would like this, but it fails in MinIO
```

Comment:
could gate these conditions on `endpoint_url.is_some()`?

Reply:
Not really, different object stores can run in different hostnames. But I think, for now at least, the defaults seem to be OK

Reply:
Right but for S3, endpoint_url will be None, so we can opt-in to the extra checks then?